docs/usage/channels.rst

```diff
@@ -0,0 +1,452 @@
+:term:`subscriber`.
+
+The plugin's :meth:`put_subscriber_history <ChannelsPlugin.put_subscriber_history>` can
+be used to fetch this history and put it into a subscriber's :term:`event stream`.
+
+.. literalinclude:: /examples/channels/put_history.py
+    :language: python
+
+
+.. note::
+    The publication of the history happens sequentially, one channel at a time, one
+    event at a time. This is done to ensure correct ordering of the events and to not
+    fill up a subscriber's backlog, resulting in dropped history entries. Should the
+    amount of entries exceed the maximum backlog size, the execution will wait until
+    previous events have been processed.
+
+    .. seealso::
+        `Managing backpressure`
+
+
+The ``Subscriber``
+------------------
+
+.. py:currentmodule:: litestar.channels.subscriber
+
+The :class:`Subscriber` manages an individual :term:`event stream`, provided to it by
+the plugin, representing the sum of events from all channels the subscriber has
+subscribed to.
+
+It can be considered the endpoint of all events, while the backends act as the source,
+and the plugin as a router, being responsible for supplying events gathered from the
+backend into the appropriate subscriber's streams.
+
+In addition to being an abstraction of an :term:`event stream`, the :class:`Subscriber`
+provides different methods to handle this stream:
+
+:meth:`iter_events <Subscriber.iter_events>`
+    An asynchronous generator, producing one event from the stream at a time, waiting
+    until the next one becomes available
+
+:meth:`start_in_background <Subscriber.start_in_background>`
+    Starts a :class:`asyncio.Task` which runs in the background, consuming the
+    :term:`event stream` and invoking a provided callback with the received events
+
+:meth:`run_in_background <Subscriber.run_in_background>`
+    A context manager, wrapping
+    :meth:`start_in_background <Subscriber.start_in_background>`. Upon exit, it will
+    attempt a graceful shutdown of the running task, waiting for all currently enqueued
+    events in the stream to be processed. Should the context be left with an error, the
+    task will be cancelled instead.
+
+    This should be the preferred method of running the background task, since it ensures
+    the task's lifetime is handled correctly
+
+    .. tip::
+        It's possible to force the task to stop immediately, by passing ``join=False`` to
+        :meth:`run_in_background <Subscriber.run_in_background>`, which will
+        lead to the cancellation of the task. By default this only happens when the context is
+        left with an exception.
+
+
+.. important::
+    The :term:`events <event>` in the :term:`event streams <event stream>` are always
+    bytes; When calling :meth:`ChannelsPlugin.publish`, data will be serialized before
+    being sent to the backend
+
+
+Consuming the event stream
++++++++++++++++++++++++++++
+
+There are two general methods of consuming the :term:`event stream`:
```

Comment:
Three?

Reply:
Not really. You can either iterate over it or use a background task.